stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -96,6 +96,11 @@ julia> diag(A, 2)
 ```
 """
 Diagonal(A::AbstractMatrix) = Diagonal(diag(A))
+Diagonal{T}(A::AbstractMatrix) where T = Diagonal{T}(diag(A))
```

Comment:
Thank you for your contribution and welcome to the julia community.

Add some tests?